addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/p1telegram/P1TelegramParser.java

```diff
@@ -0,0 +1,411 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.internal.device.p1telegram;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemObject;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemObjectFactory;
+import org.openhab.binding.dsmr.internal.device.p1telegram.P1Telegram.TelegramState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link P1TelegramParser} is a class that will read P1-port data create a full P1
+ * telegram.
+ *
+ * Data can be parsed in chunks. If a full P1 telegram is received, listeners are notified
+ *
+ * @author M. Volaart - Initial contribution
+ * @author Hilbrand Bouwkamp - Removed asynchronous call and some clean up
+ */
+@NonNullByDefault
+public class P1TelegramParser {
+
+    /**
+     * State of the parser
+     */
+    private enum State {
+    /** Wait for the '/' character */
+    WAIT_FOR_START,
```

Comment:
Formatting?

Reply:
Same problem as other enum. For some reason this is how my eclipse formats it when saving the file.